jerry-ext/args/arg-transform-functions.c

```diff
@@ -0,0 +1,249 @@
+jerryx_arg_transform_number_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  jerry_value_t to_number = jerry_value_to_number (js_arg);
+
+  if (jerry_value_has_error_flag (to_number))
+  {
+    jerry_release_value (to_number);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It can not be converted to a number.");
+  }
+
+  double *dest = jerryx_arg_iterator_get_dest (iterator_p);
+  *dest = jerry_get_number_value (to_number);
+  jerry_release_value (to_number);
+
+  return jerry_create_undefined ();
+}
+
+/* Bool transform function. */
+
+static jerry_value_t
+jerryx_arg_transform_boolean_strict_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (!jerry_value_is_boolean (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a boolean.");
+  }
+
+  double *dest = jerryx_arg_iterator_get_dest (iterator_p);
+  *dest = jerry_get_boolean_value (js_arg);
+
+  return jerry_create_undefined ();
+}
+
+static jerry_value_t
+jerryx_arg_transform_boolean_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  bool to_boolean = jerry_value_to_boolean (js_arg);
+
+  bool *dest = jerryx_arg_iterator_get_dest (iterator_p);
+  *dest = to_boolean;
+
+  return jerry_create_undefined ();
+}
+
+/* String transform function. */
+
+static jerry_value_t
+jerryx_arg_transform_string_strict_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (!jerry_value_is_string (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a string.");
+  }
+
+
+  jerry_length_t dest_len = (jerry_length_t) iterator_p->c_arg_p->extra_info;
+  char *dest = jerryx_arg_iterator_get_dest (iterator_p);
+
+  jerry_size_t size = jerry_string_to_char_buffer (js_arg,
+                                                   (jerry_char_t *) dest,
+                                                   dest_len);
+
+  if (size == 0)
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "The size of the buffer is not large enough.");
+  }
+
+  return jerry_create_undefined ();
+}
+
+static jerry_value_t
+jerryx_arg_transform_string_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  jerry_value_t to_string = jerry_value_to_string (js_arg);
+
+  if (jerry_value_has_error_flag (to_string))
+  {
+    jerry_release_value (to_string);
+
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It can not be converted to a string.");
+  }
+
+
+  jerry_length_t dest_len = (jerry_length_t) iterator_p->c_arg_p->extra_info;
```

Comment:
My bad, it should be jerry_size_t (according to the signature of jerry_string_to_char_buffer).